lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -234,8 +234,7 @@ designated_controller_event_cb(pcmk_ipc_api_t *controld_api,
     }
 
     reply = (const pcmk_controld_api_reply_t *) event_data;
-    out->message(out, "dc", reply->host_from);
-    data->rc = pcmk_rc_ok;
+    data->rc = out->message(out, "dc", reply->host_from);
```

Comment:
A message's return value should indicate only whether the message could be successfully output. We can set `data->rc` directly based on whether `host_from` is `NULL`.

If I follow correctly, this would change crmadmin's exit status from `CRM_EX_OK` to `CRM_EX_INDETERMINATE` ("Could not determine status")? If so, I think `CRM_EX_NOSUCH` ("No such object") might be better, since we could definitively determine that there is no DC. Alternatively, we could create a new rc and exit code for "DC not elected yet".